smartsim/ml/data.py

```diff
@@ -391,24 +402,24 @@ def __iter__(self) -> t.Iterator[t.Tuple[np.ndarray, np.ndarray]]:
     def init_samples(self, init_trials: int = -1) -> None:
         """Initialize samples (and targets, if needed).
 
-        A new attempt to download samples will be made every ten seconds, for ``init_trials``
-        times.
+        A new attempt to download samples will be made every ten seconds,
+        for ``init_trials`` times.
         :param init_trials: maximum number of attempts to fetch data
         :type init_trials: int
         """
         self._client = Client(self.address, self.cluster)
 
         num_trials = 0
         max_trials = init_trials or -1
-        while not len(self) and num_trials != max_trials:
+        while not self and num_trials != max_trials:
             self._update_samples_and_targets()
             self.log(
                 "DataLoader could not download samples, will try again in 10 seconds"
             )
             time.sleep(10)
             num_trials += 1
 
-        if not len(self):
+        if not self:
```

Comment:
see previous comment

Reply:
ditto on default behavior for truthiness evaluation.